kerfuffle/archiveentry.h

```diff
@@ -42,24 +43,26 @@ class Archive::Entry : public QObject
     Q_PROPERTY(QString method MEMBER m_method)
     Q_PROPERTY(QString version MEMBER m_version)
     Q_PROPERTY(QDateTime timestamp MEMBER m_timestamp)
-    Q_PROPERTY(bool isDirectory MEMBER m_isDirectory)
+    Q_PROPERTY(bool isDirectory MEMBER m_isDirectory WRITE setIsDirectory)
     Q_PROPERTY(QString comment MEMBER m_comment)
     Q_PROPERTY(bool isPasswordProtected MEMBER m_isPasswordProtected)
 
 public:
 
-    Entry(Entry *parent);
+    Entry(Entry *parent, QString fullPath = QString(), QString rootNode = QString());
```

Comment:
Make it explicit and with a default null parent:
`explicit Entry(Entry *parent = Q_NULLPTR, QString fullPath = QString(), QString rootNode = QString());`